### PR TITLE
Implement Kubectl Log

### PR DESCRIPTION
**Pull request by brendandburns**
Part of #1072

### CODE REVIEW

File: extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlLog.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import io.kubernetes.client.extended.kubectl.exception.KubectlException;
+import io.kubernetes.client.openapi.ApiClient;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+import io.kubernetes.client.openapi.models.V1Pod;
+import java.io.IOException;
+import java.io.InputStream;
+
+public class KubectlLog extends Kubectl.ResourceAndContainerBuilder<V1Pod, KubectlLog>
+    implements Kubectl.Executable<V1Pod> {
```
Comment (yue9944882) [inline extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlLog.java:25]: `Kubectl.Executable<InputStream>` will make it returning an input stream so `stream()` can be removed

File: extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlLog.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  private InputStream result;
+
+  KubectlLog(ApiClient client) {
+    super(client, V1Pod.class);
+  }
+
+  public InputStream stream() {
```
Comment (yue9944882) [inline extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlLog.java:32]: > Kubectl.Executable<InputStream> will make it returning an input stream so stream() can be removed

^^^

File: extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlLog.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import io.kubernetes.client.openapi.ApiClient;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+import io.kubernetes.client.openapi.models.V1Pod;
+import java.io.IOException;
+import java.io.InputStream;
+
+public class KubectlLog extends Kubectl.ResourceAndContainerBuilder<V1Pod, KubectlLog>
+    implements Kubectl.Executable<V1Pod> {
+  private InputStream result;
```
Comment (yue9944882) [inline extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlLog.java:26]: move this to a method-local variable?

File: extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlLog.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    try {
+      result = logs.streamNamespacedPodLog(this.name, ns, this.container);
+    } catch (ApiException | IOException ex) {
+      throw new KubectlException(ex);
+    }
+    return pod;
+  }
+
+  private void validate() throws KubectlException {
+    if (name == null || name.length() == 0) {
```
Comment (yue9944882) [inline extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlLog.java:52]: how about `Strings.isNullOrEmpty(name)`?

File: extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlLog.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+      throw new KubectlException("missing name!");
+    }
+    if (container == null || name.length() == 0) {
```
Comment (yue9944882) [inline extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlLog.java:55]: s/name.length()/container.length()/, alternatively `Strings.isNullOrEmpty(container)`?

File: extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlLog.java (hunk 1 above)
Comment (brendandburns) [reply extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlLog.java]: done.

File: extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlLog.java (hunk 2 above)
Comment (brendandburns) [reply extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlLog.java]: done.

File: extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlLog.java (hunk 3 above)
Comment (brendandburns) [reply extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlLog.java:27]: removed.

File: extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlLog.java (hunk 4 above)
Comment (brendandburns) [reply extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlLog.java]: done.

File: extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlLog.java (hunk 5 above)
Comment (brendandburns) [reply extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlLog.java]: done

**Comment by brendandburns** [comment]
Comments addressed, please re-check.

**Pull request merged**